cli/formatter/format.go

```diff
@@ -0,0 +1,788 @@
+package formatter
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"sort"
+	"strconv"
+	"strings"
+	"unicode/utf8"
+
+	"github.com/apex/log"
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/jedib0t/go-pretty/v6/text"
+	"gopkg.in/yaml.v2"
+)
+
+// MakeOutput returns formatted output depending on specified output format and formatting options.
+func MakeOutput(data string, outputFormat Format, opts *Opts) string {
```

Comment:
We don't have tests for the public API.

Reply:
Done

Reply:
This is a poor test that does not cover all formatting cases.

```Go

func TestFormatter_MakeOutputFormat(t *testing.T) {
	cases := []struct {
		outputFormat formatter.Format
		panic        bool
	}{
		{formatter.DefaultFormat, false},
		{formatter.YamlFormat, false},
		{formatter.LuaFormat, false},
		{formatter.TableFormat, false},
		{formatter.TTableFormat, false},
		{2023, true},
	}

	for _, c := range cases {
		t.Run(fmt.Sprint(c.outputFormat), func(t *testing.T) {
			formatterOpts := &formatter.Opts{
				TransposeMode: false,
				NoGraphics:    false,
				ColWidthMax:   0,
				TableDialect:  formatter.DefaultTableDialect,
			}
			if c.panic {
				f := func() {
					_ = formatter.MakeOutput("--- ['']\n...\n",
						c.outputFormat, formatterOpts)
				}
				assert.PanicsWithValue(t, "Unknown render case", f)
			} else {
				formatter.MakeOutput("--- ['']\n...\n", c.outputFormat, formatterOpts)
			}
		})
	}
}
```

Reply:
Added